### PR TITLE
Add translation mechanism

### PR DESCRIPTION
**Pull request by yan12125**
The second step of https://github.com/lxde/qterminal/issues/298, depends on https://github.com/lxde/lxqt-l10n/pull/181

### CODE REVIEW

File: lib/qtermwidget.cpp [hunk 1]
```diff
@@ -246,6 +246,15 @@ void QTermWidget::init(int startnow)
     m_layout->setMargin(0);
     setLayout(m_layout);
 
+#ifdef TRANSLATIONS_DIR
+    // translations
+    QString fname = QString("qtermwidget_%1.qm").arg(QLocale::system().name().left(5));
+    m_translator = new QTranslator(this);
+    qDebug() << "TRANSLATIONS_DIR: Loading translation file" << fname << "from dir" << TRANSLATIONS_DIR;
+    qDebug() << "load success:" << m_translator->load(fname, TRANSLATIONS_DIR, "_");
```
Comment (palinek) [inline lib/qtermwidget.cpp]: 1. isn't [this overload of load()](http://doc.qt.io/qt-5/qtranslator.html#load-1) more robust?
2. shouldn't we also consider the XDG_DATA_DIRS env. variable?

Comment (palinek) [inline lib/qtermwidget.cpp]: 3. the translator should be installed only if the `load()` succeeds, not?

Comment (yan12125) [inline lib/qtermwidget.cpp]: > shouldn't we also consider the XDG_DATA_DIRS env. variable?

These lines are copied from qterminal, which targets Mac OS X as well. I'll check how Apple bundles (*.app) cooperate with XDG standards.

**Comment by yan12125** [comment]
To implement ```$XDG_DATA_DIRS```, I'd like to move to LXQt::Translator to prevent re-inventing wheels. Is it a good idea? Currently QTermWidget depends on qt5-base only. I'm not sure whether it's fine to bring more dependencies for other qtermwidget users.

Or maybe LXQt::Translator can be split from liblxqt, just like KI18N.

**Comment by palinek** [comment]
> I'm not sure whether it's fine to bring more dependencies for other qtermwidget users.

I think most users want qtermwidget "standalone".

**Comment by yan12125** [comment]
Thanks @palinek. I've just updated the implementation, borrowing some codes from ```XdgDirs::dataDir()```

**Comment by yan12125** [comment]
Thanks for reviewing!

**Pull request merged**